x/staking/genesis.go

```diff
@@ -46,20 +47,18 @@ func InitGenesis(ctx sdk.Context, keeper Keeper, data types.GenesisState) (res [
 		keeper.SetDelegation(ctx, delegation)
 	}
 
-	sort.SliceStable(data.UnbondingDelegations[:], func(i, j int) bool {
-		return data.UnbondingDelegations[i].CreationHeight < data.UnbondingDelegations[j].CreationHeight
-	})
 	for _, ubd := range data.UnbondingDelegations {
 		keeper.SetUnbondingDelegation(ctx, ubd)
-		keeper.InsertUnbondingQueue(ctx, ubd)
+		for _, entry := range ubd.Entries {
+			keeper.InsertUBDQueue(ctx, ubd, entry.CompletionTime)
```

Comment:
Each ` entry.CompletionTime` is unique right? Otherwise, it's adding duplicates?

Reply:
> Each entry.CompletionTime is unique right

correct